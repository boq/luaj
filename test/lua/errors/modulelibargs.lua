package.path = "?.lua;test/lua/errors/?.lua"
require 'args'

-- arg type tests for module library functions

-- require
banner('require')
checkallpass('require',{{'math','coroutine','package','string','table'}},true)
checkallerrors('require',{{anumber}},'not found')
checkallerrors('require',{{anil,aboolean,afunction,atable}},'bad argument')

-- package.loadlib
-- [NOT SUPPORTED] banner('package.loadlib')
-- checkallpass('package.loadlib',{{'foo'},{'bar'}},true)
-- checkallerrors('package.loadlib',{notastring},'bad argument')

-- [REMOVED] package.seeall
-- banner('package.seeall')
-- checkallpass('package.seeall',{sometable})
-- checkallerrors('package.seeall',{notatable},'bad argument')

