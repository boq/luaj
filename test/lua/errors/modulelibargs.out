====== require ======
--- checkallpass
-		require('math')	table
-		require('coroutine')	table
-		require('package')	table
-		require('string')	table
-		require('table')	table
--- checkallerrors
-		require(1.25)	...not found...
--- checkallerrors
-		require(nil)	...bad argument...
-		require(true)	...bad argument...
-		require(<function>)	...bad argument...
-		require(<table>)	...bad argument...
